src/bridge/jsHelpers.ts

```diff
@@ -362,97 +420,104 @@ export const makeScanAccounts =
               derivationMode,
               currency,
             });
-            const showNewAccount = shouldShowNewAccount(
-              currency,
-              derivationMode
-            );
-            const stopAt = isIterableDerivationMode(derivationMode) ? 255 : 1;
-            const startsAt = getDerivationModeStartsAt(derivationMode);
 
-            log(
-              "debug",
-              `start scanning account process. MandatoryEmptyAccountSkip ${mandatoryEmptyAccountSkip} / StartsAt: ${startsAt} - StopAt: ${stopAt}`
-            );
+            let res: Result;
+            let account: Account | null | undefined;
 
-            for (let index = startsAt; index < stopAt; index++) {
-              log("debug", `start to scan a new account. Index: ${index}`);
+            if (getAddressesFromPubkey) {
+              // 1. get addresses from pubkey
+              const addresses: string[] = await getAddressesFromPubkey(
+                result.publicKey
+              );
 
-              if (finished) {
-                log("debug", `new account scanning process has been finished`);
-                break;
-              }
+              // 2. stepAccount for each address
+              for (let index = 0; index < addresses.length; index++) {
```

Comment:
So this only allows to derivate ONE account only? I mean, it only will derivate on `/0'` (account path) but then allows to have multiple address on it? 🤔 

I have a feeling we need to find a way to externalize the logic of "iteration" and the current account scan of `/n'` is one existing and default implementation. because we should avoid the `if` of this proposal imho.

but indeed it may not be that trivial. I would like to try helping on a second iteration on top of this.

Reply:
> So this only allows to derivate ONE account only? I mean, it only will derivate on /0' (account path) but then allows to have multiple address on it?

Yup, Hedera spec exactly... 1 pubkey --> n LL accounts, that's the new use-case that we don't support right now.

> I have a feeling we need to find a way to externalize the logic of "iteration" and the current account scan of /n' is one existing and default implementation. because we should avoid the if of this proposal imho.
> 
> but indeed it may not be that trivial. I would like to try helping on a second iteration on top of this.

I agree I don't like having 2 loop mechanisms in parallel, just depending on that `if`... this is kindof a first working proposal to just meet the new use-case of Hedera and likes.

If we want to externalize the logic of iteration, maybe we can provide the current default one and the new Hedera one as standalone functions of some sort, then families would provide one or the other, and then here in `jsHelpers` we would use a `perFamily` mechanism. Does this match what you had in mind?